internal/controllers/rest_controller.go

```diff
@@ -0,0 +1,62 @@
+package controllers
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/handlers"
+	"github.com/gorilla/mux"
+	"github.com/tevino/abool"
+	"github.com/ydataai/authentication-service/internal/clients"
+	"github.com/ydataai/authentication-service/internal/server"
+	"github.com/ydataai/go-core/pkg/common/logging"
+)
+
+// RESTController defines rest controller
+type RESTController struct {
+	serverConfig *server.Configuration
+	oidcConfig   *clients.OIDCConfiguration
```

Comment:
i think the properties that we need from this configuration should be in the rest configuration.

i know we can have issues with this, if someone uses the object without configuring the same endpoints, but that is theirs problem.
And we can have an assertion in the main to ensure that both properties have the same value, if required.